devenv/scripts/deploy_e2e_tester.sh

```diff
@@ -43,11 +44,11 @@ POD_LOGS_RESULT=$(run_invoke $CLUSTER_NAME $CLUSTER_RESOURCE_GROUP 'kubectl logs
 echo $POD_LOGS_RESULT
 
 
-if [ $FINALSTATUS != "0" ]
+if [[ $FINALSTATUS == "0" ]]
 then
-  echo "TEST FAILED"
-else
   echo "TEST PASSED"
+  exit 0
 fi
 
-exit $FINALSTATUS
\ No newline at end of file
+echo "TEST FAILED"
```

Comment:
It's a better pattern. There was a bug with the script beforehand that made it pass if $FINALSTATUS wasn't set (equal to ""). The pass case should be the explicit case that's being tested for to avoid unforeseen things.

Reply:
makes sense!